examples/tests/v3/test_v3_http_consumer.py

```diff
@@ -0,0 +1,94 @@
+"""
+This module provides test cases and example usage for the HTTP consumer in version 3.
+"""
+
+from pathlib import Path
+from typing import Generator
+
+import pytest
+import requests
+
+from pact.v3 import Pact
+
+
+@pytest.fixture
+def pact() -> Generator[Pact, None, None]:
+    """
+    Set up HTTP Pact Consumer.
+
+    This fixtures sets up the HTTP Pact consumer and the pact it has with a
+    provider. The consumer defines the expected http interactions it will receive from
+    the provider, and the Python test suite verifies that the correct actions
+    are taken.
+
+    For each test case, the consumer defines the following:
+
+    ```python
+
+        pact_dir = Path(Path(__file__).parent.parent.parent / "pacts")
+        pact = Pact("v3_http_consumer", "v3_http_provider")
+        yield pact.with_specification("V3")
+        pact.write_file(pact_dir)
+
+    ```
+    Returns a generator that yields the pact object and writes the pact file
+    """
+    pact_dir = Path(Path(__file__).parent.parent.parent / "pacts")
+    pact = Pact("v3_http_consumer", "v3_http_provider")
+    yield pact.with_specification("V3")
+    pact.write_file(pact_dir)
+
+
+# Example 1: GET request for existing user
+def test_get_user(pact: Pact) -> None:
+    """
+    Test the GET request for retrieving user information.
+
+    This test defines the expected interaction for a GET request to retrieve
+    user information. It sets up the expected request and response from the
+    provider and verifies that the response status code is 200.
+    """
+    # When setting up the expected response, the consumer should only define
+    # what it needs from the provider (as opposed to the full schema). Should
+    # the provider later decide to add or remove fields, Pact's consumer-driven
+    # approach will ensure that interaction is still valid.
+
+    expected_response_code = (
+        200  # In this example, we only care about the response code
+    )
+    (
+        pact.upon_receiving(
+            "a request for user information"
+        )  # Defining the interaction
+        .given("user exists")  # Defining the provider state
+        .with_request(method="GET", path="/users/1")
+        .will_respond_with(200)
+    )
```

Comment:
Could we test the response body as well?

I recall the existing examples showcasing how consumer tests only verify the _minimal_ response required. I.e., the provider might be responding with superfluous data which the consumer does not care about, and Pact should be testing only the what is strictly necessary for the consumer to work, as opposed to the entire provider payload.